### PR TITLE
Secrets from keyvault

### PR DESCRIPTION
**Pull request by DJMogens**
Final preparations for deploying a development version of the application to a server. 

I have added some stuff onto the dev branch already, but i noticed that when deploying the docker image, it wouldn't work without the firebase json file. I added the step to pull the credentials from azure key vault instead, but this needs credentials for azure. For deploying and using docker we will use .env files to authenticate to azure with a service principal ( more specifically the service principal SagaLabs-webmanager).

Review the Readme and see if you can get it to work with Az cli. I don't want to clutter the readme with how to install az cli, but here is a guide:

https://learn.microsoft.com/en-us/cli/azure/install-azure-cli. 

You should then be able to use az login, and the app should work again, provided you have the correct permissions to view our key vault. 

Jenkins.sagalabs.dk is now setup to make a instance of this application on admindev.sagalabs.dk, whenever there is a new push to dev. When we merge this, it should come up automatically

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -53,6 +53,16 @@ pipeline {
             }
         }
 
+        stage('Get .env file') {
+            steps {
+                withCredentials([file(credentialsId: 'env-file-webmanager', variable: 'ENV_FILE_PATH')]) {
+                    sh '''
+                        cp "${ENV_FILE_PATH}" .env
+                    '''
```
Comment (Tinggaard) [inline Jenkinsfile:61]: When is `ENV_FILE_PATH` set?

File: sagalabs/sagalabs.py [hunk 2]
```diff
@@ -13,22 +15,36 @@
...
 from sagalabs.db import get_db
 
-#Firebase Service Account
-cred = credentials.Certificate("secrets/FirebasePrivateKey.json")  # Path to your service account key JSON file
-firebase_admin.initialize_app(cred)
+# Azure Key Vault
+
+keyVaultUri = "https://sagalabskeyvault.vault.azure.net/"
+
+credential = DefaultAzureCredential()
```
Comment (Tinggaard) [inline sagalabs/sagalabs.py:22]: Does the dockerimage authenticate using some credentials from the ENV aswell?

File: Jenkinsfile (hunk 1 above)
Comment (DJMogens) [reply Jenkinsfile:61]: The with credentials id takes a credentials that is stored in the Jenkins credentials store and saves it as a variable in the pipeline. I set the variable and use it directly after in the cp command

File: sagalabs/sagalabs.py (hunk 2 above)
Comment (DJMogens) [reply sagalabs/sagalabs.py:22]: Yeah. The defaultazurecredential method is rather powerfull and tries several different ways of authenticating. You can be logged in with both the environment and cli, but environment will be the first to be tried. From the docs:

The following credential types if enabled will be tried, in order:

EnvironmentCredential
WorkloadIdentityCredential
ManagedIdentityCredential
AzureDeveloperCliCredential
SharedTokenCacheCredential
VisualStudioCredential
VisualStudioCodeCredential
AzureCliCredential
AzurePowerShellCredential
InteractiveBrowserCredential


https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

**Pull request merged**